src/EnergyPlus/PhaseChangeModeling/HysteresisModel.cc

```diff
@@ -0,0 +1,344 @@
+//
+// (4) Use of EnergyPlus(TM) Name. If Licensee (i) distributes the software in stand-alone form
+//     without changes from the version obtained under this License, or (ii) Licensee makes a
+//     reference solely to the software portion of its product, Licensee must refer to the
+//     software as "EnergyPlus version X" software, where "X" is the version number Licensee
+//     obtained under this License and may not use a different name for the software. Except as
+//     specifically required in this Section (4), Licensee shall not use in a company name, a
+//     product name, in advertising, publicity, or other promotional activities any name, trade
+//     name, trademark, logo, or other designation of "EnergyPlus", "E+", "e+" or confusingly
+//     similar designation, without the U.S. Department of Energy's prior written consent.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+// IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+#include <ObjexxFCL/Array1D.hh>
+
+#include <DataHeatBalance.hh>
+#include <EnergyPlus.hh>
+#include <PhaseChangeModeling/HysteresisModel.hh>
+#include <DataIPShortCuts.hh>
+#include <UtilityRoutines.hh>
+#include <InputProcessor.hh>
+
+namespace EnergyPlus {
+
+namespace HysteresisPhaseChange {
+
+	bool getHysteresisModels( true );
+	int numHysteresisModels = 0;
+	std::vector< HysteresisPhaseChange > hysteresisPhaseChangeModels;
+
+	HysteresisPhaseChange * HysteresisPhaseChange::factory( const std::string objectName ) {
+		if ( getHysteresisModels ) {
+			readAllHysteresisModels();
+			getHysteresisModels = false;
+		}
+		for ( auto & hm : hysteresisPhaseChangeModels ) {
+			if ( hm.name == objectName ) {
+				return &hm;
+			}
+		}
+		// because of the passive linking between materials and material property objects,
+		// we don't know ahead of time for sure whether we will have a material property
+		// so we can't return fatal here if it isn't found, just leave it null
+		return nullptr;
+	}
+
+	Real64 HysteresisPhaseChange::getEnthalpy( Real64 T, Real64 Tc, Real64 tau1, Real64 tau2 ) {
+		// Looks up the enthalpy on the characteristic curve defined by the parameters Tc, tau1, and tau2,
+		// and the position on that curve defined by T.
+		Real64 eta1 = ( this->totalLatentHeat / 2 ) * exp( -2 * abs( T - Tc ) / tau1 );
```

Comment:
tau1 and tau2 will always be assigned a meaningful value from the input object.  The input object includes \minimum>0 so these values should be guaranteed to always have meaningful, >0, values.  I am cleaning up the logic at the top of this function while addressing this.